### DISCUSSION TITLE
no gtagOption for debug_mode?

### DISCUSSION THREAD

**Discussion by ezbeazy** [codler/react-ga4#98] [Q&A] [answered]
I don't see the gtag debug_mode option mapped to any parameter for ReactGA.initialize(). It also doesn't appear to be related to options.testMode, and I don't know what testMode is for since their is no description. Will the below not work? How can we use debug_mode? and what is options.testMode for?

```
    export const initGA = (ga4Id: string, devMode: boolean) => {
        ReactGA.initialize(ga4Id, {dev_mode: devMode});
    };
```

```
if (
    !window.GA_ANALYTICS &&
    googleAnalyticsTrackingCode
    ){
      
      if (apiHost.includes('localhost:'){
        initGA(googleAnalyticsTrackingCode, true);
      } else {
        initGA(googleAnalyticsTrackingCode, false);
      }

     window.GA_ANALYTICS = true;
}
```

**Comment by ezbeazy** [comment]
upon further inspection it appears gatagOptions does allow the flexibility to add options not explicitly mapped and I should be able to use the dev_mode option described in the "[Monitor events in DebugView](https://support.google.com/analytics/answer/7201382?sjid=10106802497287070450-NA#zippy=%2Cgoogle-tag-gtagjs)" help article. Is that correct?

**Comment by KrisLau** [comment]
Did you ever figure it out?

**Comment by anindya-dhruba** [comment, marked as answer]
This works for me:
```
ReactGA4.initialize(GA4_MEASUREMENT_ID, {
  gaOptions: {
    debug_mode: true
  }
});
```
You can view the tracking data in the **Debug View** page.

**Note:** To disable debug mode, remove the debug_mode. Setting this to `false` won't work.

**Reply by codler** [reply]
Additional example https://github.com/codler/react-ga4/discussions/52#discussioncomment-6487947
